app/ui/src/app/platform/types/integration/integration.models.ts

```diff
@@ -23,24 +23,34 @@ export const ERROR = 'Error';
 
 export type IntegrationStatus = 'Pending' | 'Published' | 'Unpublished' | 'Error';
 
-export interface Integration extends BaseEntity {
+export interface IntegrationOverview extends BaseEntity {
+  version?: number;
+  tags: Array<string>;
   description?: string;
+  draft: boolean;
+  deployments?: Array<DeploymentOverview>;
+  currentStatus: IntegrationStatus;
+  targetState: IntegrationStatus;
+  statusMessage?: string;
+  deploymentVersion?: number;
+}
+
+export type IntegrationOverviews = Array<IntegrationOverview>;
+
+export interface Integration extends IntegrationOverview {
```

Comment:
The only thing I'm concerned about here is that this isn't a real relationship.  And the `Integration` object no longer exposes any state/status fields.  The main impact I'm thinking about here is more for saving an integration which requires the whole object.  I guess though we could just trim that out as needed when it becomes a problem.